silver/tests/spec/test_payment_method.py

```diff
@@ -1,27 +1,308 @@
+        response = self.client.get(url, format='json')
+
+        data = response.data
+        payment_processor = reverse('payment-processor-detail',
+                                    kwargs={'processor_name': 'someprocessor'},
+                                    request=response.wsgi_request)
+        data['payment_processor'] = payment_processor
+
+        response = self.client.put(url, data=data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_detail(self):
+        payment_method = self.create_payment_method(customer=self.customer,
+                                                    state=PaymentMethod.States.Unverified)
+
+        url = reverse('payment-method-detail', kwargs={
+            'customer_pk': self.customer.pk,
+            'payment_method_id': payment_method.pk
+        })
+
+        response = self.client.get(url, format='json')
+        data = response.data
+        data['state'] = PaymentMethod.States.Enabled
+
+        response = self.client.put(url, data=data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, data)
+
+    def test_get_listing_no_customer(self):
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': sys.maxint
+        })
+
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_get_detail_no_customer(self):
+        url = reverse('payment-method-detail', kwargs={
+            'customer_pk': sys.maxint,
+            'payment_method_id': 0
+        })
+
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_get_detail_no_payment_method(self):
+        url = reverse('payment-method-detail', kwargs={
+            'customer_pk': self.customer.pk,
+            'payment_method_id': sys.maxint
+        })
+
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_post_listing_no_customer(self):
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': sys.maxint
+        })
+
+        response = self.client.post(url, data={
+            'payment_processor': processor_url,
+            'state': PaymentMethod.States.Uninitialized
+        }, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_post_listing_incomplete_body_1(self):
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': 0
+        })
+
+        response = self.client.post(url, data={}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
Here as well. `response.data` Probably elsewhere too, but won't spam the review with this.